lib/facter/framework/config/fact_groups.rb

```diff
@@ -41,6 +45,15 @@ def get_group_ttls(group_name)
       ttls_to_seconds(ttls[group_name])
     end
 
+    def get_fact(fact_name)
+      return @facts_ttls[fact_name] if @facts_ttls[fact_name]
+
+      fact = @facts_ttls.select { |k, v| break v if fact_name =~ /^#{k}\..*/ }
```

Comment:
`k, v` are not very suggestive, maybe use `key`, `value`.
Also that break there is a bit hard to read overall. Do you need just the first value that matches `/^#{k}\..*/`?

Reply:
Yes, we take the first match. `@facts_ttls` is sorted so the children come first (os.macosx vs os)

Reply:
You could use detect instead of select. It returns the first element for which the block returns true.